### ISSUE TITLE
Since update 0.0.16, audio is heavily distorted

### ISSUE THREAD

**Issue by jimj316** [crmarsh/discord-linux-bugs#208]
Discord was working fine yesterday, but today it updated to 0.0.16, and audio - both transmit and receive - is coming through extremely crackled and distorted, with an audible echo. This is so bad that it is almost impossible to hear what anyone is saying, and so Discord is effectively unusable for me until this is fixed.

I am on Debian Stretch (testing), using KDE Plasma 5.8.6. My ALSA is at version 1.0.27+1.

**Comment by Lerentis** [comment]
Having the same issue in Arch with GNOME. Had to ditch the client for now and use the website..

ALSA: k4.10.13-1-ARCH
GNOME: 3.24.2
Kernel: 4.10.13-1
Electron: v1.4.16

**Comment by crmarsh** [comment]
I've heard other reports from folks claiming more decryption errors in the latest build, and one instance where people said they couldn't hear anyone else correctly until they had spoken themselves. I wonder if these are related.

**Comment by sliceofcode** [comment]
Also experiencing this issue.

Kernel: 4.10.13-1-ARCH
Discord Canary: 0.0.16-1 (from AUR)

```
pulseaudio 10.0-3
pulseaudio-alsa 2-3
alsa-lib 1.1.3-1
alsa-plugins 1.1.1-1
alsa-utils 1.1.3-2
pulseaudio-alsa 2-3
```

**Comment by Lerentis** [comment]
@crmarsh i don't know if it helps to narrow down this issue but i don't have it in a regular call. only in voice channel communication

**Comment by alpertoygar** [comment]
i have this problem too but the voice becomes normal after 1-2 minutes

**Comment by winneon** [comment]
I am experiencing this as well under Arch with pulseaudio installed. Because of another bug preventing me from using alsa, I'm forced to use pulseaudio. Like @alpertoygar said, it seems to go away after a minute or two. Disconnecting and reconnecting also seems to fix the issue. I hope this gets resolved within a reasonable time frame.

**Comment by qalshidi** [comment]
I have this same problem. I am usually too annoyed to go past 2 minutes but I'll check if this makes it work for me too.

**Comment by gourdcaptain** [comment]
This bug still exists in heavy crackling in calls for me on Stable 0.0.2 and Canary 0.0.24. Deafening and undeafening myself usually fixes it for the call.

**Comment by benrob0329** [comment]
Says I have over 2k ms ping, but everything else on my system is fine.

**Comment by qalshidi** [comment]
I have checked it did subside with me after a while. Unsure what is causing this.

**Comment by gourdcaptain** [comment]
I think I _may_ have found a workaround and way to reproduce the audio static bug. I tested it about five times in a private call, but given the unreliable nature of the bug who knows:
Workaround: Run Discord with the following environment variable setting line "env PULSE_LATENCY_MSEC=60 discord" (or discord-canary as appropriate).
Reproduction: Set "env PULSE_LATENCY_MSEC=3 discord" (or discord-canary).

**Comment by OvermindDL1** [comment]
> I think I may have found a workaround and way to reproduce the audio static bug. I tested it about five times in a private call, but given the unreliable nature of the bug who knows:
Workaround: Run Discord with the following environment variable setting line "env PULSE_LATENCY_MSEC=60 discord" (or discord-canary as appropriate).
Reproduction: Set "env PULSE_LATENCY_MSEC=3 discord" (or discord-canary).

That makes it sound like Discord is not getting the update rate from the sound system and feeding sound data to it at that rate?

**Comment by gourdcaptain** [comment]
I'll admit this was entirely a blind guess based off this being something that used to work with similar problems with the old binary Linux Skype client.

**Comment by gazny** [comment]
I have this problem too, it doesn't affect my input but my output gets distorted on all channels, it distorts my in-game audio,music, and calls/channels.
What would cause it to go away after being in a call for multiple minutes?

**Comment by gourdcaptain** [comment]
FYI, it may be worth if you're using the PULSE_LATENCY_MSEC workaround to find the lowest value that works for you to minimize the effects. 30 seems to work for me.

**Comment by Brottweiler** [comment]
Few days ago I disabled [timer-based scheduling](https://wiki.archlinux.org/index.php/PulseAudio/Troubleshooting#Glitches.2C_skips_or_crackling) and I've had voice call a couple of times now with no issues.

**Comment by winneon** [comment]
@Brottweiler Yes this is what fixed it for me as well. Pulseaudio in general just seems to be much more work than what it's worth.

**Comment by gourdcaptain** [comment]
Hey, I'm going to ask - everyone who's got the problem, what version of PulseAudio are you using? Arch just pushed 11.0, and some... very casual testing didn't cause the issue to pop up. Might be a version of it causing the issue. I'm guessing everyone on Arch had 10.0, but I'm not sure what the people with Ubuntu have.

**Comment by winneon** [comment]
I do have 10.0 on Arch. When I do a system upgrade I'll re-enable timer-based scheduling on 11.0 and see if the problem persists.

**Comment by OvermindDL1** [comment]
I'm using pulseaudio 8.0 and I have no distortion issues.  Perhaps I should update...  >.>

**Comment by gourdcaptain** [comment]
Okay, just got the distortion with PulseAudio 11 after a day of not getting it. *sigh*

**Comment by avallete** [comment]
I had the same problem with crakling audio, becoming normal after 1-5 minutes.

I've use the solution given by @Brottweiler and that resolved the bug (3 day testing, no problem).

For those who didn't want to read the documentation, on Debian:
```
# Probably must sudo
sed 's/^load-module module-udev-detect$/load-module module-udev-detect tsched=0/' -i /etc/pulse/default.pa
```

**Comment by JShorthouse** [comment]
Run discord with `PULSE_DELAY_MSEC=30` aka `PULSE_DELAY_MSEC=30 discord` from the terminal.

If that fixes the problem then you can edit the discord .desktop file (which should be at `/usr/share/applications/discord.desktop` and change the Exec line to have `env PULSE_LATENCY_MSEC=30 ` before the path.

**Comment by Lukypie** [comment]
I'm still having this problem also with Discord Canary and Discord Stable.
Wtf? Only arch users? I am using manjaro

**Comment by qalshidi** [comment]
I am using Manjaro as well and as I've stated above the fix worked if you use the environment variable

**Comment by Lukypie** [comment]
Really annoying, even with the "fix" the problem persists after minutes, then it fixes itself. But doing that every time you are connecting is really annoying.

**Comment by Bafford-the-dour** [comment]
I tried Ubuntu, Fedora and Manjaro. The problem seems to be the same everywhere.

**Comment by andyklimczak** [comment]
Something that seems to work for me is to, start discord and join a voice channel (all audio should be fuzzy at this point). Then quit and restart discord, when you reconnect you'll still be in the channel and (hopefully) the sound will be fixed.

Sometimes it takes a few restarts, but the key is to be in a voice channel when restarting discord.

**Comment by urmamasllama** [comment]
Running antergos. I've got the same issue but pulse delay fix doesn't seem to be working for me

**Comment by falsechicken** [comment]
I also have this issue under Ubuntu Mate 17.10. Its incredibly annoying. Discord is the only app that does this so far.

**Comment by Bafford-the-dour** [comment]
So the same issue occured to me earlier today while not having discord open, just firefox. By the way, the issue now also occurs on the browser version of discord for me. I'm starting to think that this might be a pulseaudio problem, that only discord seems to trigger for the most part. It's pulseaudio after all...

**Comment by Bafford-the-dour** [comment]
I found some thread from all the way back in 2011 that seems to describe our problem...
https://bbs.archlinux.org/viewtopic.php?id=116172

**Comment by Bafford-the-dour** [comment]
Editing /etc/pulse/default.pa seems to have done the trick for me. I changed the line
`load-module module-udev-detect`
to
`load-module module-udev-detect tsched=0`
According to a user on the Arch forum, this will make the scheduling process more processor intensive and apparently it may break audio in WINE too. If I find out anyhting else, I'll report back.

**Comment by Brottweiler** [comment]
@Bafford-the-dour Literally what I said in August in this specific issue... I've not had a single problem since I disabled timer based scheduling in August.
